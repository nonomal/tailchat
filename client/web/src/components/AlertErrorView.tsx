import React, { useState } from 'react';
import { Alert, Button } from 'antd';
import clsx from 'clsx';
import { t } from 'tailchat-shared';

/**
 * 用于接口错误显示的组件
 * @deprecated 请使用 ErrorView
 */
export const AlertErrorView: React.FC<{
  error: Error;
}> = React.memo(({ error }) => {
  const [show, setShow] = useState(false);

  const description = (
    <div>
      <span>{String(error.message)}</span>
      <Button
        className={clsx({
          'opacity-0': show,
        })}
        type="link"
        onClick={() => setShow(true)}
      >
        {t('显示详情')}
      </Button>
      {show && <pre>{String(error.stack)}</pre>}
    </div>
  );

  return (
    <Alert
      className="w-full h-full select-text"
      type="error"
      message={String(error.name)}
      description={description}
    />
  );
});
AlertErrorView.displayName = 'AlertErrorView';
