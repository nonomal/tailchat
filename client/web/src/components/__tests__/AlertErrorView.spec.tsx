import { render } from '@testing-library/react';
import React from 'react';
import { AlertErrorView } from '../AlertErrorView';

describe('AlertErrorView', () => {
  test('render', () => {
    const wrapper = render(<AlertErrorView error={new Error('Dummy Error')} />);
    expect(wrapper.container).toMatchSnapshot();
  });
});
