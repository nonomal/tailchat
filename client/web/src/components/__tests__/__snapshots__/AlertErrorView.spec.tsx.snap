// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertErrorView render 1`] = `
<div>
  <div
    class="ant-alert ant-alert-error ant-alert-with-description ant-alert-no-icon w-full h-full select-text"
    data-show="true"
    role="alert"
  >
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        Error
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <span>
            Dummy Error
          </span>
          <button
            class="ant-btn ant-btn-link"
            type="button"
          >
            <span>
              显示详情
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
